### ISSUE TITLE
Dpo vs Sft test truthfulqa

### ISSUE THREAD

**Issue by Minami-su** [huggingface/trl#1320]
Dpo vs Sft test
Test model: Qwen-1_8B-Chat,Mistral-7B-Instruct-v0.2
Datasets: jondurbin/truthy-dpo-v0.1
Dpo framework: [Sakura-SOLAR-DPO](https://github.com/KyujinHan/Sakura-SOLAR-DPO)
Sft framework: alpaca-lora
Dpo default lr:  1e-5 gradstep 16
Sft default lr:  1e-4 gradstep 1
all model load_in_4bit

hf-causal (pretrained=Qwen-1_8B-Chat), limit: None, provide_description: False, num_fewshot: 0, batch_size: 64
|    Task     |Version|Metric|Value |   |Stderr|
|-------------|------:|------|-----:|---|-----:|
|truthfulqa_mc|      1|mc1   |0.2460|±  |0.0151|
|             |       |mc2   |0.3878|±  |0.0145|

hf-causal (pretrained=Qwen-1_8B-Chat-dpo epoch 3), limit: None, provide_description: False, num_fewshot: 0, batch_size: 64
|    Task     |Version|Metric|Value |   |Stderr|
|-------------|------:|------|-----:|---|-----:|
|truthfulqa_mc|      1|mc1   |0.2558|±  |0.0153|
|             |       |mc2   |0.4221|±  |0.0148|

hf-causal (pretrained=Qwen-1_8B-Chat-truth-dpov2epoch 6), limit: None, provide_description: False, num_fewshot: 0, batch_size: 64
|    Task     |Version|Metric|Value |   |Stderr|
|-------------|------:|------|-----:|---|-----:|
|truthfulqa_mc|      1|mc1   |0.2570|±  |0.0153|
|             |       |mc2   |0.4247|±  |0.0148|

hf-causal (pretrained=Qwen-1_8B-Chat-truth-sft-epoch6), limit: None, provide_description: False, num_fewshot: 0, batch_size: 64
|    Task     |Version|Metric|Value |   |Stderr|
|-------------|------:|------|-----:|---|-----:|
|truthfulqa_mc|      1|mc1   |0.2632|±  |0.0154|
|             |       |mc2   |0.4464|±  |0.0154|

hf-causal (pretrained=Qwen-1_8B-Chat-sft-epoch3-dpo-epoch1), limit: None, provide_description: False, num_fewshot: 0, batch_size: 64
|    Task     |Version|Metric|Value |   |Stderr|
|-------------|------:|------|-----:|---|-----:|
|truthfulqa_mc|      1|mc1   |0.2644|±  |0.0154|
|             |       |mc2   |0.4475|±  |0.0152|

hf-causal (pretrained=Qwen-1_8B-Chat-truth-sft-epoch3), limit: None, provide_description: False, num_fewshot: 0, batch_size: 64
|    Task     |Version|Metric|Value |   |Stderr|
|-------------|------:|------|-----:|---|-----:|
|truthfulqa_mc|      1|mc1   |0.2619|±  |0.0154|
|             |       |mc2   |0.4619|±  |0.0154|

hf-causal (pretrained=Qwen-1_8B-Chat-sft-epoch3-dpo-epoch6), limit: None, provide_description: False, num_fewshot: 0, batch_size: 64
|    Task     |Version|Metric|Value |   |Stderr|
|-------------|------:|------|-----:|---|-----:|
|truthfulqa_mc|      1|mc1   |0.2644|±  |0.0154|
|             |       |mc2   |0.4677|±  |0.0158|

hf-causal (pretrained=Mistral-7B-Instruct-v0.2-sft-epoch3), limit: None, provide_description: False, num_fewshot: 0, batch_size: 96
|    Task     |Version|Metric|Value |   |Stderr|
|-------------|------:|------|-----:|---|-----:|
|truthfulqa_mc|      1|mc1   |0.4370|±  |0.0174|
|             |       |mc2   |0.6232|±  |0.0156|(base model already trained this datasets,so this is overfitting)

hf-causal (pretrained=Mistral-7B-Instruct-v0.2), limit: None, provide_description: False, num_fewshot: 0, batch_size: 64
|    Task     |Version|Metric|Value |   |Stderr|
|-------------|------:|------|-----:|---|-----:|
|truthfulqa_mc|      1|mc1   |0.4688|±  |0.0175|
|             |       |mc2   |0.6317|±  |0.0156|

hf-causal (pretrained=Mistral-7B-Instruct-v0.2-dpov2epoch1 lr:5e-7), limit: None, provide_description: False, num_fewshot: 0, batch_size: 96 finally loss 0.6531
|    Task     |Version|Metric|Value |   |Stderr|
|-------------|------:|------|-----:|---|-----:|
|truthfulqa_mc|      1|mc1   |0.4725|±  |0.0175|
|             |       |mc2   |0.6326|±  |0.0156|

hf-causal (pretrained=Mistral-7B-Instruct-v0.2-dpov1epoch1), limit: None, provide_description: False, num_fewshot: 0, batch_size: 96 finally loss 0.03
|    Task     |Version|Metric|Value |   |Stderr|
|-------------|------:|------|-----:|---|-----:|
|truthfulqa_mc|      1|mc1   |0.4859|±  |0.0175|
|             |       |mc2   |0.6545|±  |0.0154|

hf-causal (pretrained=Mistral-7B-Instruct-v0.2-dpo-epoch6), limit: None, provide_description: False, num_fewshot: 0, batch_size: 64 inally loss 0.0001
|    Task     |Version|Metric|Value |   |Stderr|
|-------------|------:|------|-----:|---|-----:|
|truthfulqa_mc|      1|mc1   |0.4871|±  |0.0175|
|             |       |mc2   |0.6677|±  |0.0154|

Conclusion:
1.Sft is suitable for the training of epoch3 on untrained data sets, and the over-fitting effect will become worse
2.Dpo is used in all scenarios. When the model has been trained on the sft on the corresponding data set, it can continue to use dpo promotion with the same data.
3. The more loss training rounds of dpo, the better, and the lower the loss, the less overfitting
